### ISSUE TITLE
Search results do not highlight if results are not an exact match

### ISSUE THREAD

**Issue by nathanstitt** [documentcloud/documentcloud#123]
For instance if someone searches for "Solutions" and the search returns matches based off the word "Solution" being in the document, the text tab does not highlight anything since it can only deal with exact matches.

The "previous" and "next" buttons also page through invisible words and do not appear to be doing anything.

Perhaps we should recognize when the search doesn't have an exact match on the Text tab and put a message up to that effect instead of displaying the buttons.

**Comment by reefdog** [comment]
It seems Solr is smarter than our front end. Its stemming configuration matches a search for `facility` to both `facility` and `facilities`, etc. But the viewer's interface for navigating search results does a simple [case-insensitive regex for terms to highlight](https://github.com/documentcloud/document-viewer/blob/3eb4bfd76ada80bd6013f4161321c75d719b832f/public/javascripts/DV/helpers/search.js#L76-L81). When it encounters a page with no literal results (because the morphology is different), it gets stuck.

**Hard requirement:** Solr and the front end should be in sync.

**Ideally:** The front end is as smart as Solr. (I.e., it knows how to locate and highlight the morphed terms, not just the literals.) This will be a big lift, as it will require rewriting the viewer's search interface.

**Suggested compromise:** Configure Solr's stemming so that it does literal case-insensitive matches only, like the interface. This is a smaller and dumber result set, but it won't break out interface.

**Problems with compromise:** This interface problem only affects the viewer, yet a configuration change affects all searches (incl. API and workspace). Even if we can configure Solr to perform one (limited) search for the viewer and another (advanced) search for the workspace and API, the differing result set would be unacceptable. I think?

**Comment by knowtheory** [comment]
As far as i'm concerned the viewer is in the wrong/broken.

I'm not entirely sure what the best course of action going forward is going to be.  We need to either get character positions for highlighting out of solr's response, for the purpose of highlighting, or we need to display solr's representation of each page with highlights directly in each page (currently we're pulling each page from cloudfront, and then running a secondary process to highlight it).

**Comment by reefdog** [comment]
I agree with you, but if we aren't able to rewrite the interface for a while, then the current system is more broken than temporarily dumbing down Solr is wrong. Would you consider that as an interim solution? After which yeah, either of those solutions (having the search API send back the text with results pre-highlighted, ala the workspace search, or have it send back the character positions) are cool by me. Latter seems more flexible.
